src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,44 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's Contact.
+ * Parent class for Address, Phone and Email
+ */
+public class Contact {
+
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Creates a given contact
+     */
+    public Contact(String contact, boolean isPrivate){
+        this.isPrivate = isPrivate;
+        contact = contact.trim();
+        this.value = contact;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
```

Comment:
At any point if you decide to override the equals method in sub classes, keep in mind to override the hashCode() too.